### PR TITLE
Support structs with apply functions in db::apply

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]
@kidder

File: src/DataStructures/DataBox/DataBox.hpp [hunk 1]
```diff
@@ -1414,12 +1414,26 @@ constexpr const Type& get_item_from_box(const DataBox<TagList>& box,
...
+                Args...>> = nullptr>
+  static constexpr auto apply(F&& /*f*/, const DataBox<BoxTags>& box,
+                              Args&&... args) {
+    return F::apply(::db::get<Tags>(box)..., std::forward<Args>(args)...);
+  }
+
+  template <typename F, typename BoxTags, typename... Args,
+            Requires<not is_apply_callable_v<
+                F, const std::add_lvalue_reference_t<db::item_type<Tags>>...,
+                Args...>> = nullptr>
```
Comment (kidder) [inline src/DataStructures/DataBox/DataBox.hpp:1436]: document that you can do this below, adding the example?

**Review by kidder — APPROVED** [review]
Looks good!

**Comment by kidder** [comment]
@nilsdeppe please look at

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Travis failures are timeouts, all tests passed.

**Pull request merged**